datafusion/sql/src/set_expr.rs

```diff
@@ -36,8 +40,17 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
                 right,
                 set_quantifier,
             } => {
-                let left_plan = self.set_expr_to_plan(*left, planner_context)?;
-                let right_plan = self.set_expr_to_plan(*right, planner_context)?;
+                let left_plan = self.set_expr_to_plan(*left.clone(), planner_context)?;
```

Comment:
Can we please try and avoid this clone ? It looks like only the left and right spans are needed so we could copy them out rather than the entire set expr?